core/scripts/cli/sponsor/create.js

```diff
@@ -6,27 +7,32 @@ const create = async (web3, artifacts, emp) => {
   const ExpandedERC20 = artifacts.require("ExpandedERC20");
   const { toWei, fromWei, toBN } = web3.utils;
 
-  const scalingFactor = toBN(toWei("1"));
   // TODO: Understand why we need a .rawValue in one case but not the other.
-  const totalPositionCollateral = toBN((await emp.totalPositionCollateral()).rawValue.toString());
-  const totalTokensOutstanding = toBN((await emp.totalTokensOutstanding()).toString());
+  const totalPositionCollateral = BigNumber((await emp.totalPositionCollateral()).rawValue.toString());
+  const totalTokensOutstanding = BigNumber((await emp.totalTokensOutstanding()).toString());
   if (totalTokensOutstanding.isZero()) {
     // When creating the globally first position, we wouldn't have a GCR. Therefore, creating that position is a
     // different flow that isn't currently part of this tool.
     console.log("Error: This tool does not currently support creating the chosen market's first position");
     return;
   }
-  const gcr = totalPositionCollateral.mul(scalingFactor).divRound(totalTokensOutstanding);
 
   const input = await inquirer.prompt({
     message: "How many tokens to create?",
     name: "tokensCreated",
     validate: value => value > 0 || "Number of tokens must be positive"
   });
-  const tokens = toWei(input["tokensCreated"]);
-  const collateralNeeded = toBN(tokens)
-    .mul(gcr)
-    .divRound(scalingFactor);
+  // Use BigNumber.js to so that we can set ROUNDING_MODE to round ceiling. We need to do this to make sure we send
+  // enough collateral to create the requested tokens.
+  BigNumber.set({ ROUNDING_MODE: 2 });
+  const scalingFactor = BigNumber(toWei("1"));
+  const tokens = BigNumber(toWei(input["tokensCreated"]));
+  const gcr = totalPositionCollateral.times(scalingFactor).div(totalTokensOutstanding);
+  const collateralNeeded = tokens
+    .times(gcr)
+    .div(scalingFactor)
+    .integerValue()
```

Comment:
does `integerValue`() just take the integer part of a float returned by `div`?

Reply:
Yep. But the div already rounds using ceil mode.